lib/processor.js

```diff
@@ -3,6 +3,27 @@
  * @author Brandon Mills
  */
 
+/**
+ * @typedef {import('eslint/lib/shared/types').LintMessage} Message
+ *
+ * @typedef {Object} ASTNode
+ * @property {string} type
+ * @property {string} [lang]
+ *
+ * @typedef {ASTNode & {children: ASTNode[]}} ASTParentNode
+ *
```

Comment:
Since #177 removed the need for the parent node, can you confirm that we're good to get rid of the `ASTParentNode` type? If so, hopefully you can commit this and the other suggestion inline to save effort.

```suggestion
```